pkg/cloud/aws/actuators/machine_scope.go

```diff
@@ -128,4 +129,7 @@ func (m *MachineScope) Close() {
 	if err != nil {
 		klog.Errorf("[machinescope] failed to store provider status for machine %q in namespace %q: %v", m.Machine.Name, m.Machine.Namespace, err)
 	}
+
+	// control plane node count is a status of the cluster. update that incase a new control plane node was created.
+	m.Scope.Close()
```

Comment:
This was removed because it causes races between the machine and cluster actuator.

Reply:
Dropping this approach